audio/out/ao_pipewire.c

```diff
@@ -626,13 +637,110 @@ static void add_device_to_list(struct ao *ao, uint32_t id, const struct spa_dict
     ao_device_list_add(list, ao, &(struct ao_device_desc){name, description});
 }
 
+static void hotplug_registry_global_cb(void *data, uint32_t id,
+                                       uint32_t permissions, const char *type,
+                                       uint32_t version, const struct spa_dict *props)
+{
+    struct ao *ao = data;
+    struct priv *priv = ao->priv;
+
+    if (!is_sink_node(type, props))
+        return;
+
+    pw_thread_loop_lock(priv->loop);
+    struct id_list *item = talloc_size(NULL, sizeof(*item));
+    if (item) {
+        item->id = id;
+        spa_list_init(&item->node);
+        spa_list_append(&priv->hotplug.sinks, &item->node);
+    } else {
+        MP_WARN(ao, "Could not allocate list element\n");
```

Comment:
all of the `talloc_` functions abort rather than returning NULL on OOM

Reply:
Nice. Changed.